src/main/java/userinterface/UserInterface.java

```diff
@@ -1,7 +1,496 @@
+     *               with an RU.
+     */
+
+    private static void createCarrierOnRu(String ruName, Scanner input) {
+        RatType ratType;
+        List<RfPort> rfPorts;
+        FrequencyBand freqBand;
+        double transPower;
+
+        // get RAT type
+        ratType = chooseRatType(input);
+
+        //get RF Ports
+        rfPorts = chooseRfPorts(ratType, input);
+
+        //get Frequency band
+        freqBand = chooseFreqBand(ratType, input);
+
+        //get transmitting power
+        transPower = getTransPower(input);
+
+        if (ruName != null) {
+            mediator.createCarrierOnRu(rfPorts, freqBand, transPower, ruName);
+        } else {
+            mediator.createCarrier(rfPorts, freqBand, transPower);
+        }
+
+    }
+
+
+    /**
+     * Helper method that create a carrier on existing RU based on user-choose
+     * RAT type, RF ports, frequency band and transmitting power.
+     *
+     * @param ruName The name of the radio unit to create carrier on.
+     */
+
+    private static void createCarrierAndRu(String ruName, Scanner input) {
+        Vendor vendor;
+        RatType ratType;
+        List<RfPort> rfPorts;
+        FrequencyBand freqBand;
+        double transPower;
+
+        //get vendor for radio unit
+        vendor = chooseVendor(input);
+
+        //get RAT type
+        ratType = chooseRatType(input);
+
+        //get RF Ports
+        rfPorts = chooseRfPorts(ratType, input);
+
+        //get Frequency band
+        freqBand = chooseFreqBand(ratType, input);
+
+        //get transmitting power
+        transPower = getTransPower(input);
+
+        mediator.createCarrierAndRu(rfPorts, freqBand, transPower, ruName, vendor, ratType);
+
+    }
+
+    /**
+     * Helper method that get the transmitting power for carrier from user
+     * Returns the transmitting power user entered
+     *
+     * @param input Scanner instance to get user input.
+     * @return The transmitting power, as a double.
+     */
+    private static double getTransPower(Scanner input) {
+        double transPower;
+        System.out.println("Please enter a number for transmitting Power:\n");
+        do {
+            String transPowerText = input.next();
+            try {
+                transPower = Double.parseDouble(transPowerText);
+                break;
+            } catch (NumberFormatException e) {
+                System.out.println("'" + transPowerText + "'" + " is not a valid number, Try again!");
+            }
+        } while (true);
+        return transPower;
+    }
+
+
+    /**
+     * Helper method that get the vendor from user.
+     * Returns the vendor user selected.
+     *
+     * @param input Scanner instance to get user input.
+     * @return A vendor from the Vendor enum.
+     */
+    private static Vendor chooseVendor(Scanner input) {
+        Vendor vendor = null;
+        System.out.println("Please choose a Vendor for Radio Unit:\n"

```

Comment:
I think a future improvement we could add is to leverage the enum here - there should be a way to list all of the enum values so that adding new vendors won't require a change here. Similarly below, when we parse the user's input to a vendor enum, we could probably add some method to the enum class that takes the id (e.g. number) and returns the appropriate enum.

Reply:
I agree this would make more sense and aim to remove a lot of the code for these methods currently in the UserInterface class, I just didn't have enough time to get around to making that change. But I'll adapt to this in a new commit.